virttest/utils_libvirt/libvirt_ceph_utils.py

```diff
@@ -0,0 +1,154 @@
+                                                               key_file)
+        else:
+            device_source = "rbd:%s:mon_host=%s" % (ceph_disk_name, ceph_mon_ip)
+        logging.debug("device source is: %s", device_source)
+        # clean up image file if exists
+        cmd = ("rbd -m {0} {1} info {2} && rbd -m {0} {1} rm "
+               "{2}".format(ceph_mon_ip, key_opt, ceph_disk_name))
+        process.run(cmd, ignore_status=True, shell=True, verbose=True)
+
+        #Create necessary image file if not exists
+        if img_file is None:
+            img_file = os.path.join(data_dir.get_tmp_dir(),
+                                    "%s_test.img" % vm.name)
+            # Create an local image and make FS on it.
+            disk_cmd = ("qemu-img create -f %s %s %s" %
+                        (device_format, img_file, storage_size))
+            process.run(disk_cmd, ignore_status=False, shell=True)
+        # Convert the image to remote ceph storage
+        disk_path = ("rbd:%s:mon_host=%s" %
+                     (ceph_disk_name, ceph_mon_ip))
+        if ceph_client_name and ceph_client_key:
+            disk_path += (":id=%s:key=%s" %
+                          (ceph_auth_user, ceph_auth_key))
+        rbd_cmd = ("rbd -m %s %s info %s 2> /dev/null|| qemu-img convert -O"
+                   " %s %s %s" % (ceph_mon_ip, key_opt, ceph_disk_name,
+                                  device_format, img_file, disk_path))
+        process.run(rbd_cmd, ignore_status=False, shell=True, verbose=True)
+
+        # Disk related config
+        disk_src_dict = {"attrs": {"protocol": "rbd",
+                                   "name": ceph_disk_name},
+                         "hosts":  [{"name": ceph_mon_ip,
+                                     "port": ceph_host_port}]}
+        # Create disk xml with given config
+        vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+        disk_xml = Disk(type_name='network')
+        disk_xml.device = device
+        disk_xml.target = {"dev": device_target, "bus": device_bus}
+        driver_dict = {"name": "qemu", "type": device_format}
+        disk_xml.driver = driver_dict
+        disk_source = disk_xml.new_disk_source(**disk_src_dict)
+        if disk_auth_dict:
+            logging.debug("disk auth dict is: %s" % disk_auth_dict)
+            disk_xml.auth = disk_xml.new_auth(**disk_auth_dict)
+        disk_xml.source = disk_source
+        logging.debug("new disk xml is: %s", disk_xml)
+        if not keep_raw_image_as:
+            if hotplug:
+                attach_option = params.get("virt_device_attach_option", "--config")
+                virsh.attach_device(vm.name, disk_xml.xml,
+                                    flagstr=attach_option, ignore_status=False, debug=True)
+            else:
+                vmxml.add_device(disk_xml)
+                vmxml.sync()
+    else:
+        cmd = ("rbd -m {0} {1} info {2} && rbd -m {0} {1} rm "
```

Comment:
we already have some ceph related utils in virttest.ceph, can we replace such kind of shell command lines with common utilities to improve readability?

Reply:
I can not understand what do you mean 'readiablity' here.
If we look at real execution logic in test cases, that is actual case that we use rbd command to execute that.
Doesn't it native and original command make readability friendly and easy to debug?

Reply:
I agree with yingshun as the common function will show more clear what you are doing and can make any update more easily, and maintain easily. Please use the common function as much as possible, thanks

Reply:
> I agree with yingshun as the common function will show more clear what you are doing and can make any update more easily, and maintain easily. Please use the common function as much as possible, thanks

@kylazhang 
let us find seconds offline to discuss why one rule can not apply all over world

Reply:
updated